packages/design-picker/src/components/mshots-image/index.tsx

```diff
@@ -35,57 +45,76 @@ export function mshotsUrl( url: string, options: MShotsOptions, count = 0 ): str
 
 const MAXTRIES = 10;
 
+// https://stackoverflow.com/a/60458593
+const useMshotsUrl = ( src: string, options: MShotsOptions ) => {
+	const [ loadedSrc, setLoadedSrc ] = useState( '' );
+	const [ count, setCount ] = useState( 0 );
+
+	useEffect( () => {
+		const img = new Image();
+		const srcUrl = mshotsUrl( src, options, count );
+		img.src = srcUrl;
+		img.onload = () => {
+			// Detect default image (Don't request a 400x300 image).
+			if ( img.naturalWidth !== 400 || img.naturalHeight !== 300 ) {
+				setLoadedSrc( srcUrl );
+			} else {
+				// Only refresh 10 times
+				count < MAXTRIES &&
+					// Triggers a target.src change with increasing timeouts
+					setTimeout( () => setCount( count + 1 ), count * 500 );
+			}
+		};
+	}, [ src, count ] );
```

Comment:
`options` should be added to the dependency array here. I think React just does an object equivalence check here, which is fine for us because it looks like we're treating the mshot options as immutable in the code. We could be extra paranoid about this and add the `readonly` TS annotation to each of the `MShotsOptions` properties? I haven't seen us do that a lot in Calypso though, and much of our code assumes immutable objects, so 🤷